core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/neo/WorkerPromiseManager.scala

```diff
@@ -0,0 +1,26 @@
+package edu.uci.ics.amber.engine.architecture.worker.neo
+
+import com.twitter.util.Promise
+import edu.uci.ics.amber.engine.architecture.messaginglayer.ControlOutputPort
+import edu.uci.ics.amber.engine.architecture.worker.neo.promisehandlers.PauseHandler
+import edu.uci.ics.amber.engine.common.ambermessage.WorkerMessage.ExecutionPaused
+import edu.uci.ics.amber.engine.common.ambertag.neo.VirtualIdentity.ActorVirtualIdentity
+import edu.uci.ics.amber.engine.common.promise.{PromiseInvocation, PromiseManager, WorkflowPromise}
+
+class WorkerPromiseManager(
+    val selfID: ActorVirtualIdentity,
+    val controlOutputPort: ControlOutputPort,
+    val pauseManager: PauseManager,
+    val dataProcessor: DataProcessor
+) extends PromiseManager(selfID, controlOutputPort)
+    with PauseHandler {
```

Comment:
I dont think `with PauseHandler` is a good way. This way we will have to add all control message handlers here as dependency injection. That shouldnt be the way

Reply:
I think it is the only way we can separate all the handling logic. Any other way you can think of?